lib/chef/resource.rb

```diff
@@ -811,7 +812,10 @@ def updated=(true_or_false)
     #
     # @return [String] The DSL name of this resource.
     def self.dsl_name
-      convert_to_snake_case(name, 'Chef::Resource')
```

Comment:
Thing is that dsl_name is pretty shitty.  The actual dsl name is in the `#delcared_type` on the insetance which is actually created.  I only see a couple of places where this gets used internally, and I doubt its getting much use externally.  This seems like a fine time to just make this go away.  If I'm wrong we can always back out, but keeping this method around doesn't make any sense to me.  It makes a bit more sense as a name for the semantic of `#declared_type` but its on the class, not the instance, so we can't ever make it work right.